modules/nrfxlib/nrf_802154/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if (CONFIG_NRF_802154_RADIO_DRIVER OR CONFIG_NRF_802154_SERIALIZATION)
+
+if (NOT CONFIG_NRF_802154_SL_OPENSOURCE AND NOT CONFIG_SOC_SERIES_NRF53X)
+  target_sources(nrf-802154-platform
```

Comment:
the `nrf-802154-platform`is defined in `zephyr` itself, and then sources are added here in `sdk-nrf` but include directories and link libraries are specified in `sdk-nrfxlib`.

Would it be better to have a variable to control use of the platform lib, similar to what is done for radio and serilization ?

Reply:
The problem with those source files (namely the `nrf_802154_hp_timer.c` and `nrf_802154_lp_timer.c`) is that those files are never used in vanilla zephyr, therefore I thought that it is better add them only in NCS glue. I have made sure that https://github.com/zephyrproject-rtos/zephyr/blob/51efef4ff24b04a523bb7da9a11b6484e1248133/modules/hal_nordic/CMakeLists.txt will always be evaluated before this file by adding the `depends` statement to https://github.com/nrfconnect/sdk-nrfxlib/blob/4b15172170e628d0059d754532f72117bf02a63a/zephyr/module.yml.

Reply:
thanks for the explanation.